src/main/scala/cromwell/engine/db/slick/SlickDataAccess.scala

```diff
@@ -636,6 +637,62 @@ class SlickDataAccess(databaseConfig: Config) extends DataAccess {
       } }
   }
 
+  override def addCallFailureEvent(workflowId: WorkflowId, executionKey: ExecutionDatabaseKey,
+                                           failure: FailureEventEntry)(implicit ec: ExecutionContext): Future[Unit] = {
+    val action = for {
+      workflowExecution <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
+      execution <- executionKey match {
+        case ExecutionDatabaseKey(callFqn, Some(index), attempt) => dataAccess.executionsByWorkflowExecutionUuidAndCallFqnAndShardIndexAndAttempt(workflowId.toString, callFqn, index, attempt).result.head
+        case ExecutionDatabaseKey(callFqn, None, attempt) => dataAccess.executionsByWorkflowExecutionUuidAndCallFqnAndAttempt(workflowId.toString, callFqn, attempt).result.head
+      }
+
+      _ <- dataAccess.failureEventsAutoInc +=
+        new FailureEvent(
+          workflowExecution.workflowExecutionId.get,
+          Option(execution.executionId.get),
+          StringUtils.abbreviate(failure.failure, 255),
+          new Timestamp(failure.timestamp.getMillis))
+    } yield ()
+
+    runTransaction(action)
+  }
+
+  override def addWorkflowFailureEvent(workflowId: WorkflowId, failure: FailureEventEntry)(implicit ec: ExecutionContext): Future[Unit] = {
+    val action = for {
+      workflowExecution <- dataAccess.workflowExecutionsByWorkflowExecutionUuid(workflowId.toString).result.head
+
+      _ <- dataAccess.failureEventsAutoInc +=
+        new FailureEvent(
+          workflowExecution.workflowExecutionId.get,
+          None,
+          StringUtils.abbreviate(failure.failure, 255),
```

Comment:
That's the current length of the message text column.  I will both enlarge that value and turn this into a descriptive constant.